KubeArmor/core/kubeUpdate.go

```diff
@@ -541,12 +541,18 @@ func (dm *KubeArmorDaemon) WatchK8sPods() {
 				if ownerRef != nil {
 					podOwnerName = ownerRef.Name
 					if ownerRef.Kind == "ReplicaSet" {
-						deploymentName := K8s.GetDeploymentNameControllingReplicaSet(pod.Metadata["namespaceName"], podOwnerName)
+						deploymentName := ownerRef.Name //K8s.GetDeploymentNameControllingReplicaSet(pod.Metadata["namespaceName"], podOwnerName)
```

Comment:
@DelusionalOptimist need you review here,
we need `ownerRef.Name` here  for `K8s.K8sClient.AppsV1().ReplicaSets(pod.Metadata["namespaceName"]).Get(context.Background(), deploymentName, metav1.GetOptions{})`

didnt see any errors while testibg but wanted to be sure before removing it

Reply:
@Shreyas220 if I understand correctly, this pod ownerRef would be a replica set. We actually need the deployment name, which is the owner of the replicaset itself. Thus we have this in place.
The profile naming convention we have current goes like: `kubearmor-namespace-deployment-container` and this would change it.

Reply:
so `K8s.GetDeploymentNameControllingReplicaSet()``  made sure we use the naming convention ?

The deployment name returned from `K8s.GetDeploymentNameControllingReplicaSet()` connot be used to perform .GET() and .Patch() operations

Reply:
> so K8s.GetDeploymentNameControllingReplicaSet() made sure we use the naming convention ?

Yes, the deployment name returned by `K8s.GetDeploymentNameControllingReplicaSet()` is be used to create apparmor profiles.

> The deployment name returned from K8s.GetDeploymentNameControllingReplicaSet() connot be used to perform .GET() and .Patch() operations

Not sure what do you mean :thinking:.
We have always used it to patch the deployments earlier in the method `PatchDeploymentWithAppArmorAnnotations`. In fact, I think you won't be able to patch the deployment if you use the ReplicaSet name.

Reply:
yeah found out what the issue is 
even for the infinite loop issue you mentioned below it was related to this 

so if the owner is a replica set it doesn't return any value so changed it to `ownerref.Name`

now if the owner of replicaset is a deployment we  using this `ownerref.Name` creates a problem resulting in the infinite loop

but the infinite loop only happens when we do make run not when i use container to deploy kubearmor 

cant seem to figure that one out :joy:

Reply:
this was the issue for the error below 
only reproducible when testing using `make run`